src/libstore/meson.build

```diff
@@ -0,0 +1,458 @@
+project('nix-store', 'cpp',
+  version : run_command('cat', './.version', check : true).stdout().strip(),
+  default_options : [
+    'cpp_std=c++2a',
+    # TODO(Qyriad): increase the warning level
+    'warning_level=1',
+    'debug=true',
+    'optimization=2',
+    'errorlogs=true', # Please print logs for tests that fail
+  ],
+)
+
+cxx = meson.get_compiler('cpp')
+
+deps_private = [ ]
+deps_public = [ ]
+deps_other = [ ]
+
+configdata = {
+  # TODO rename, because it will conflict with downstream projects
+  'PACKAGE_VERSION' : '"' + meson.project_version() + '"',
+  'SYSTEM': '"' + host_machine.system() + '"',
+}
```

Comment:
Probably, configdata should be an instance of `configuration_data()` so that you can use `.set_quoted()` instead of implementing this via `'"' + var + '"'` which is not precisely the height of readability.